exercises/bowling/canonical-data.json

```diff
@@ -1,117 +1,110 @@
+      "description": "rolling a spare with the two roll bonus does not get a bonus roll",
+      "rolls": [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 10, 7, 3],
+      "expected": 20
     }, {
-      "description": "should allow fill balls to be a strike",
+      "description": "strikes with the two roll bonus do not get bonus rolls",
       "rolls": [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 10, 10, 10],
       "expected": 30
     }, {
-      "description": "should be able to score a perfect game",
+      "description": "a strike with the one roll bonus after a spare in the last frame does not get a bonus",
+      "rolls": [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 7, 3, 10],
+      "expected": 20
+    }, {
+      "description": "all strikes is a perfect game",
       "rolls": [10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10],
       "expected": 300
-    }]
-  },
-  "validations": {
-    "description": [
-      "Check game rules."
-    ],
-    "cases": [{
-      "description": "should not allow rolls with negative pins",
-      "rolls": [-1],
-      "expected": "Pins must have a value from 0 to 10"
     }, {
-      "description": "should not allow rolls better than strike",
-      "rolls": [11],
-      "expected": "Pins must have a value from 0 to 10"
+      "description": "Rolls can not score negative points",
+      "rolls": [-1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+      "expected": -1
     }, {
-      "description": "should not allow two normal rolls better than strike",
-      "rolls": [5, 6],
-      "expected": "Pin count exceeds pins on the lane"
+      "description": "A roll can not score more than 10 points",
+      "rolls": [11, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+      "expected": -1
     }, {
-      "description": "should not allow two normal rolls better than strike in last frame",
-      "rolls": [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 5, 6],
-      "expected": "Pin count exceeds pins on the lane"
+      "description": "Two rolls in a frame can not score more than 10 points",
+      "rolls": [5, 6, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+      "expected": -1
     }, {
-      "description": "should not allow to take score at the beginning of the game",
+      "description": "An unstarted game can not be scored",
       "rolls": [],
-      "expected": "Score cannot be taken until the end of the game"
+      "expected": -1
     }, {
-      "description": "should not allow to take score before the game has ended",
-      "rolls": [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
-      "expected": "Score cannot be taken until the end of the game"
+      "description": "An incomplete game can not be scored",
+      "rolls": [0, 0],
+      "expected": -1
     }, {
-      "description": "should not allow rolls after the tenth frame",
+      "description": "A game with more than ten frames can not be scored",
```

Comment:
The change in the description seems to change the implication of where the error should occur.

Before, to me this sounds like: The roll should be rejected before there is any attempt to take the score.

After, it sounds like: The roll should be accepted, and onlyh when taking the score do we realize there is a failure.

And I definitely think the roll should be rejected as soon as possible.

But this is a difficult proposition when the only expectation this test file expresses is the score at the end, instead of noting where the error should occur. That's a difficult concept to express though. The first idea that comes to mind (and the idea may be crazy) is:

```
"rolls": [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
"roll_should_fail": 0,
```